app/services/dossier_field_service.rb

```diff
@@ -138,6 +139,14 @@ def sorted_ids(dossiers, procedure_presentation, gestionnaire)
 
     private
 
+    def sanitized_column(field)
+      table = field['table']
+      table = ActiveRecord::Base.connection.quote_column_name((table == 'self' ? 'dossier' : table).pluralize)
```

Comment:
Pinailliage : j'aurais bien extrait la partie `(table == 'self' ? 'dossier' : table).pluralize` pour simplifier la lecture

Reply:
Pinailliage 2 : d'ailleurs on peut n'apporter cette partie que dans un commit ultérieur il me semble, car c'est pas utilisé dans celui-ci

Reply:
> À mon avis, la Vraie Solution™ serait de remplacer self par dossier un peu partout dans la classe pour éviter d‘avoir à gérer ce cas particulier. Par contre, ça implique une migration de données et je ferais bien ça dans une PR séparée

OK pour en discuter, ça me semble intéressant comme piste.

> Pas compris, qu’est-ce qui n’est pas utilisé ?

Dans le commit `[#2579] Fix injection SQL dans le filtrage instructeur `, où tu ajoutes `#sanitized_column`, cette méthode contient l'instruction `(table == 'self' ? 'dossier' : table)`. Or, dans ce commit, on n'appelle jamais (il me semble) cette colonne avec le cas `table == 'self'` (ce qui m'a d'ailleurs interpellé lors de ma review). Donc la gestion de ce cas (et donc l'ajout de cette instruction) pourrait être rajoutée dans cette méthode dans le commit où ce cas (`table == 'self'`) est rencontré par la méthode (je crois qu'il s'agit de `[#2579] Simplify sorted_ids` ou de ` [#2579] Protect against SQL injection on column and table in filtered_ids`)